nncf/quantization/algorithms/post_training/algorithm.py

```diff
@@ -175,41 +84,14 @@ def apply(
         statistic_points: Optional[StatisticPointsContainer] = None,
         dataset: Optional[Dataset] = None,
     ) -> TModel:
-        modified_model = copy_model(model)
-        modified_model_graph = graph
-        backend = get_backend(modified_model)
-
-        for algorithm in self.first_stage_algorithms:
-            if isinstance(algorithm, SmoothQuant) and backend != BackendType.OPENVINO:
-                nncf_logger.debug(f"{backend.name} does not support SmoothQuant algorithm yet.")
-                continue
-
-            if isinstance(algorithm, ChannelAlignment) and backend != BackendType.OPENVINO:
-                nncf_logger.debug(f"{backend.name} does not support ChannelAlignment algorithm yet.")
-                continue
-
-            statistics_aggregator = StatisticsAggregatorFactory.create(modified_model, dataset)
-            algo_statistic_points = algorithm.get_statistic_points(modified_model, modified_model_graph)
-            statistics_aggregator.register_statistic_points(algo_statistic_points)
-            statistics_aggregator.collect_statistics(modified_model, modified_model_graph)
-            modified_model = algorithm.apply(
-                modified_model, modified_model_graph, statistics_aggregator.statistic_points
+        if dataset is None and len(self._pipeline.pipeline_steps) > 1:
```

Comment:
What would happen if the `len(self._pipeline.pipeline_steps)` be 1? What should the user do?

Reply:
There are two scenarios

1. When `len(pipeline_steps)` equals `1`, in this case, we can pass only a value for the `statistic_points` parameter to the `PostTrainingQuantization.apply()` method, and there's no need to pass a value for the `dataset` parameter.
   ```python
   quantized_model = PostTrainingQuantization.apply(model, graph, statistic_points=statistic_points)
   ```
   It is also possible to pass only a value for the `dataset` parameter to call this method.
   ```python
   quantized_model = PostTrainingQuantization.apply(model, graph, dataset=calibration_dataset)
   ```
2. When `len(pipeline_steps)` is greater than `1`, in this case, a value for the `dataset` parameter is always required because it is necessary to collect statistics for intermediate models. So, the `PostTrainingQuantization.apply()` method can be called the following ways
   ```python
   # Pass the statistic_points collected for the first step and the dataset
   quantized_model = PostTrainingQuantization.apply(model, graph, statistic_points, dataset)

   # Pass the dataset only
   quantized_model = PostTrainingQuantization.apply(model, graph, dataset=calibration_dataset)
   ```
I understand that this doesn't appear user-friendly. I guess this API will be improved here https://github.com/openvinotoolkit/nncf/pull/2013